iroha/src/queue.rs

```diff
@@ -42,6 +26,29 @@ pub struct Queue {
     ttl: Duration,
 }
 
+/// Queue push error
+#[derive(Error, Debug)]
+pub enum Error {
+    /// Queue is full
+    #[error("Queue is full")]
+    Full(VersionedAcceptedTransaction),
+    /// Transaction expired
+    #[error("Transaction is expired")]
+    Expired(VersionedAcceptedTransaction),
+    /// Transaction is already in blockchain
+    #[error("Transaction is already applied")]
+    InBlockchain(VersionedAcceptedTransaction),
+    /// Signature condition check failed
+    #[error("Failure during signature condition execution")]
+    SignatureCondition {
+        /// Signature condition check error
+        #[source]
+        error: Report,
+        /// Transaction
+        tx: VersionedAcceptedTransaction,
+    },
```

Comment:
Why is it needed to return the transaction back?

Reply:
In order to reuse it. Like to show its hash in logs also